zipline/algorithm.py

```diff
@@ -693,9 +697,14 @@ def symbol(self, symbol_str):
         Default symbol lookup for any source that directly maps the
         symbol to the Asset (e.g. yahoo finance).
         """
+        # If the user has not set the symbol lookup date,
+        # use the period_end as the date for sybmol->sid resolution.
+        _lookup_date = self.symbol_lookup_date if self.symbol_lookup_date \
```

Comment:
datetimes and pd.Timestamps are True, but confusingly the time component of them is False at UTC midnight:

```python
>>> dt = Timestamp('1970-01-01 00:00:00+0000', tz='UTC')
>>> bool(dt)
True
>>> midnight = Timestamp('1970-01-01 00:00:00+0000', tz='UTC').time()
>>> bool(midnight)
False
```

However, just to be extra-specially sure I'll add a `is not None` check.